system/lib/wasmfs/syscalls.cpp

```diff
@@ -444,7 +444,28 @@ static __wasi_fd_t doOpen(path::ParsedParent parsed,
     }
   }
 
-  // TODO: Return EISDIR if child is a directory and write access is requested.
+  if (auto link = child->dynCast<Symlink>()) {
+    if (flags & O_NOFOLLOW) {
+      return -ELOOP;
+    }
+    // TODO: The link dereference count starts back at 0 here. We could
+    // propagate it from the previous path parsing instead.
+    auto target = link->getTarget();
+    auto parsedLink = path::getFileFrom(parent, target);
```

Comment:
Why is the parent needed here? Shouldn't we get the absolute path from to the target? Maybe I don't understand how posix symlinks work...

Reply:
Symlinks containing relative paths are interpreted relative to their parent directories during path parsing. Otherwise what they refer to would depend on the current working directory, which would be very weird! For symlinks containing absolute paths, `parent` will end up being unused here, as you expect.